proposals/NNNN-experiment-tracking/README.md

```diff
@@ -0,0 +1,495 @@
+associated with Kubeflow Pipelines' heavy reliance on MLMD.
+
+## Motivation
+
+Currently, experiment tracking in Kubeflow is fragmented and tightly coupled to specific components. Kubeflow Pipelines
+requires users to run everything through pipelines to track experiments, while Katib's experiments are limited to
+hyperparameter tuning scenarios. This fragmentation creates several challenges:
+
+1. **Limited flexibility**: Users cannot log experiments directly from Python scripts, Jupyter Notebooks, or other
+   Kubeflow components without forcing everything through pipelines. This restriction often drives users to seek
+   solutions outside the Kubeflow ecosystem.
+1. **Fragmented experience**: Users must navigate multiple interfaces to correlate run results and evaluation data,
+   depending on which Kubeflow component they use (e.g., Pipeline runs, Katib experiments, Training Operator results).
+1. **No unified tracking**: Users cannot easily compare runs, share insights, or maintain consistent metadata across the
+   entire Kubeflow ecosystem.
+1. **Maintenance challenges**: The Kubeflow Pipelines dependency on MLMD (Machine Learning Metadata) creates technical
+   debt and limits future flexibility due to its lack of active maintenance. MLMD also overlaps with the Model Registry
+   objectives defined in the
+   [Model Registry overview](https://www.kubeflow.org/docs/components/model-registry/overview/#use-case-2-experimenting-with-different-model-weights-to-optimize-model-accuracy),
+   creating separate metadata store experiences.
+
+These limitations make it difficult to effectively manage the machine learning lifecycle across Kubeflow.
+
+### Goals
+
+1. **Kubeflow centralized experiment tracking**: Create an experiment tracking system that is independent from Kubeflow
+   Pipelines, enabling experiments to be logged and managed without requiring pipeline execution.
+1. **Transform Model Registry into a unified metadata store**: Evolve the existing Model Registry component into a
+   centralized metadata store that can handle experiments, runs, metrics, and artifacts across the entire Kubeflow
+   ecosystem.
+1. **Pipelines integration**: Integrate Kubeflow Pipelines with the centralized experiment tracking system in Model
+   Registry, allowing pipeline runs and output artifacts to be automatically logged and tracked.
+1. **Remove the MLMD dependency from Pipelines**: Decouple Kubeflow Pipelines from MLMD (Machine Learning Metadata) to
+   remove a core dependency on a project outside of the Kubeflow ecosystem.
+1. **Extend SDK capabilities for experiment tracking**: Enhance the Kubeflow SDK to support direct logging of
+   experiments, runs, metrics, and artifacts without requiring pipeline execution.
+1. **Provide MLFlow SDK compatibility**: Create an
+   [MLFlow SDK plugin](https://mlflow.org/docs/latest/ml/plugins/#storage-plugins) that allows users to leverage the
```

Comment:
I am of two minds about explicitly setting this as a goal, since the plugin is standalone. Having the MLFlow plugin as a separate enhancement might make it easier to approve, both.